evap/staff/views.py

```diff
@@ -802,11 +814,62 @@ def questionnaire_copy(request, questionnaire_id):
             return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
     else:
         questionnaire = get_object_or_404(Questionnaire, id=questionnaire_id)
+        form, formset = get_identical_form_and_formset(questionnaire)
+        return render(request, "staff_questionnaire_form.html", dict(form=form, formset=formset))
+
+
+def find_questionnaire_name(old_questionnaire):
+    timestamp = datetime.date.today()
+    new_name_de = '{} (until {})'.format(old_questionnaire.name_de, str(timestamp))
+    new_name_en = '{} (until {})'.format(old_questionnaire.name_en, str(timestamp))
+
+    if Questionnaire.objects.filter(Q(name_de=new_name_de) | Q(name_en=new_name_en)):
+        return None, None
```

Comment:
ok, right. could you try moving that error handling to the top of the view so it applies to the POST part as well for extra safety? and could you check how inlining `find_questionnaire_name` does look like then?